azext_iot/central/commands_device.py

```diff
@@ -120,3 +120,10 @@ def get_credentials(
     return provider.get_device_credentials(
         device_id=device_id, central_dns_suffix=central_dns_suffix,
     )
+
+
+def generate_device_key(
+    cmd, masterkey, device_id,
+):
+    provider = CentralDeviceProvider(cmd=cmd, app_id=None, token=None,)
```

Comment:
Since a big use case for this is offline generation of dps keys, lets remove the provider pattern in this one case